pkg/healthchecker/health_checker_windows.go

```diff
@@ -94,6 +89,18 @@ func getHealthCheckFunc(hco *options.HealthCheckerOptions) func() (bool, error)
 	return nil
 }
 
+// healthCheckEndpointOKFunc returns a function to check the status of an http endpoint
+func healthCheckEndpointOKFunc(endpoint string, timeout time.Duration) func() (bool, error) {
```

Comment:
Seems like a missed opportunity to not make this generic health check through config. IE remove the hardcoded addresses and pass that in as a json argument.

/cc @Random-Liu what do you think?

Reply:
> @Random-Liu what do you think?

Yeah, we thought about introducing the Kubernetes like "healthiness probe": Exec probe + http probe.

The only problem is that the health checker is already a plugin of NPD, and now we need another layer of plugin. :P 

Some design is needed here.

Please feel free to suggest ideas or propose a design for this, but that doesn't necessarily need to block this change.